### ISSUE TITLE
Removes space before human size output

### ISSUE THREAD

**Issue by wernight** [docker/docker#30271]
**Description**

**Steps to reproduce the issue:**
1. `$ docker images | sort -hk 7,8` 

**Describe the results you received:**
Sorts numerically but doesn't take into account the ` MB` or ` GB` (even though `-k 7,8` includes it).

**Describe the results you expected:**
Most commands, like `du -h` have no space before `MB` or `GB` and thus work directly with `sort -h`.
The workaround is an ugly hack `$ docker images | sed 's/ \(MB\|kB\|GB\)/\1/' | sort -hk 7`.

**Additional information you deem important (e.g. issue happens only occasionally):**
Common tools that have the standard `K`, `M`, `G` without space (note that they don't have `GB` but just `G` but that doesn't seem to bother `sort -h`):

 - `du -h`
 - `df -h`
 - `ls -lh`
 - `tree -h`
 - `free -h`

Common tools that have a space before it:

 - `docker images`

**Output of `docker version`:**

```
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 21:44:32 2016
 OS/Arch:      linux/amd64
```

**Comment by AkihiroSuda** [comment]
`docker images` itself is likely to support sorting
https://github.com/docker/docker/pull/25702

**Comment by wernight** [comment]
That's neat, but I still think it shouldn't try to reinvent the wheel and
keep closer to Linux standards (especially even Windows is kind of moving
towards them). It would be also a much simpler "fix".

Le jeu. 19 janv. 2017 à 10:50, Akihiro Suda <notifications@github.com> a
écrit :

**Comment by thaJeztah** [comment]
I agree that it would be nice to stick closer to other tools, and don't see a real issue in removing the space.

The package uses `kiB` / `kB` to distinguish between kibibyte and kilobyte, which (according to Wikipedia (fwiw)) is the "official" unit https://en.wikipedia.org/wiki/Kibibyte, so not sure which to use there.

comparing the outputs of `df -h` between my local macOS machine,  and an Ubuntu box (kibibyte vs kilobyte);

macOS;

```
df -h
Filesystem                           Size   Used  Avail Capacity    iused      ifree %iused  Mounted on
/dev/disk2                          2.8Ti  1.2Ti  1.6Ti    43%    4201672 4290765607    0%   /
devfs                               193Ki  193Ki    0Bi   100%        666          0  100%   /dev
...
```

Ubuntu 16.04;

```
Filesystem      Size  Used Avail Use% Mounted on
udev            982M     0  982M   0% /dev
tmpfs           201M  696K  200M   1% /run
```

@wernight could you open a pull-request for discussion in https://github.com/docker/go-units (which is where the "human friendly" output is generated?

Perhaps it's best to open separate PR's for "remove space" and changing the unit, so that they can be discussed independent from each other.

**Comment by wernight** [comment]
I did one. Strangely it's not referenced here.

**Comment by ripcurld0** [comment]
@thaJeztah can we close this ticket? or we need to re-vendor go-unit?

**Comment by wernight** [comment]
Need to vendor that go-units but didn't have time to perform that

**Comment by yongtang** [comment]
I think this issue has been fixed through #30602 and #30790?

**Comment by thaJeztah** [comment]
yes it is!
